src/main/java/com/cryptomorin/xseries/XItemStack.java

```diff
@@ -113,6 +115,8 @@ public static void serialize(@Nonnull ItemStack item, @Nonnull ConfigurationSect
             config.set("damage", item.getDurability());
         }
 
+        if (meta == null || !item.hasItemMeta()) return;
```

Comment:
I'm pretty sure `ItemStack#hasItemMeta()` will also take care of `item == null`
Plus, this should be before `item.getItemMeta()` is actually invoked.

You should also take care of that `Damageable` meta. Just move `ItemStackMeta meta =...` after setting amount and `hasItemMeta` check before it.